src/composer.rs

```diff
@@ -1,30 +1,51 @@
+use acvm::async_trait;
+
 use crate::barretenberg_structures::{Assignments, ConstraintSystem};
-use crate::crs::{CRS, G2};
-use crate::{Barretenberg, Error, FIELD_BYTES};
+use crate::crs::download_crs;
+use crate::{crs::CRS, Barretenberg, Error, FIELD_BYTES};
 
 const NUM_RESERVED_GATES: u32 = 4; // this must be >= num_roots_cut_out_of_vanishing_polynomial (found under prover settings in barretenberg)
 
+#[async_trait]
 pub(crate) trait Composer {
     fn get_circuit_size(&self, constraint_system: &ConstraintSystem) -> Result<u32, Error>;
 
     fn get_exact_circuit_size(&self, constraint_system: &ConstraintSystem) -> Result<u32, Error>;
 
-    fn compute_proving_key(&self, constraint_system: &ConstraintSystem) -> Result<Vec<u8>, Error>;
-    fn compute_verification_key(
+    async fn get_crs(&self, constraint_system: &ConstraintSystem) -> Result<CRS, Error> {
+        let num_points = self.get_circuit_size(constraint_system)?;
+
+        download_crs(num_points as usize).await
```

Comment:
Should there be an option to read from a cache? Proposed in #184

Reply:
I believe this conflicts with https://github.com/noir-lang/noir/issues/1302